File: cdm/csharp/properties/SEScalarNegative1To1.cs
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

public class SEScalarNegative1To1 : SEScalar
{
    public SEScalarNegative1To1() : base()
    {

    }

    public SEScalarNegative1To1(double value)
    {
        SetValue(value);
    }

    public override void SetValue(double val)
    {
        if (val >= 0 || val <= 1)
            value = val;
        else
            throw new System.Exception("Value must be [0-1]");
    }
}

File: src/cpp/cdm/properties/SEScalarNegative1To1.cpp
/* Distributed under the Apache License, Version 2.0.
   See accompanying NOTICE file for details.*/

#include "stdafx.h"
#include "properties/SEScalarNegative1To1.h"

SEScalarNegative1To1::SEScalarNegative1To1() : SEScalar()
{

}

void SEScalarNegative1To1::SetValue(double d)
{ 
  if (d > 1 || d < -1)
    throw CommonDataModelException("SEScalarNegative1To1 must be between [-1,1]");
  SEScalar::SetValue(d);
}

void SEScalarNegative1To1::SetValue(double d, const NoUnit& unitless)
{ 
  SEScalarNegative1To1::SetValue(d);
}

void SEScalarNegative1To1::ForceValue(double d)
{
  SEScalar::ForceValue(d);
}

void SEScalarNegative1To1::ForceValue(double d, const NoUnit& unitless)
{
  SEScalar::ForceValue(d);
}
